includes/documents/class-wcpdf-invoice.php

```diff
@@ -103,27 +103,53 @@ public function init_number() {
 			return $invoice_number;
 		}
 
-		$number_store_method = WPO_WCPDF()->settings->get_sequential_number_store_method();
-		$number_store_name = apply_filters( 'wpo_wcpdf_document_sequential_number_store', 'invoice_number', $this );
-		$number_store = new Sequential_Number_Store( $number_store_name, $number_store_method );
-		// reset invoice number yearly
-		if ( isset( $this->settings['reset_number_yearly'] ) ) {
-			$current_year = date("Y");
-			$last_number_year = $number_store->get_last_date('Y');
-			// check if we need to reset
-			if ( $current_year != $last_number_year ) {
-				$number_store->set_next( apply_filters( 'wpo_wcpdf_reset_number_yearly_start', 1, $this ) );
-			}
-		}
-
-		$invoice_date = $this->get_date();
-		$invoice_number = $number_store->increment( $this->order_id, $invoice_date->date_i18n( 'Y-m-d H:i:s' ) );
+		$number_store   = $this->get_sequential_number_store();
+		$invoice_number = $number_store->increment( $this->order_id, $this->get_date()->date_i18n( 'Y-m-d H:i:s' ) );
 
 		$this->set_number( $invoice_number );
 
 		return $invoice_number;
 	}
 
+	public function get_sequential_number_store() {
+		$reset_number_yearly = isset( $this->settings['reset_number_yearly'] ) ? true : false;
+		$method              = WPO_WCPDF()->settings->get_sequential_number_store_method();
+	
+		// reset: on
+		if( $reset_number_yearly ) {
+			$store_name   = $this->get_sequential_number_store_name( $this->get_date() );
+			$number_store = new Sequential_Number_Store( $store_name, $method );	
+	
+			if ( $number_store->is_new ) {
+				$number_store->set_next( apply_filters( 'wpo_wcpdf_reset_number_yearly_start', 1, $this ) );
+			}
+		// reset: off
+		} else {
+			$now          = new \WC_DateTime( "@".time(), new \DateTimeZone( 'UTC' ) );
+			$store_name   = $this->get_sequential_number_store_name( $now );
+			$number_store = new Sequential_Number_Store( $store_name, $method );
+	
+			if ( $number_store->is_new ) {
+				// get last year's store if available
+				$last_year         = $now->modify('-1 year');
+				$last_store_name   = $this->get_sequential_number_store_name( $last_year );
+				$last_number_store = new Sequential_Number_Store( $last_store_name, $method );	
+				if ( ! $number_store->is_new ) {
+					$number_store->set_next( apply_filters( 'wpo_wcpdf_reset_number_yearly_start', $last_number_store->get_next(), $this ) );
+				}
+			}
+		}
+	
+		return $number_store;
+	}
+	
+	public function get_sequential_number_store_name( $date ) {
+		$year       = $date->date_i18n( 'Y' );
+		$store_name = "{$this->slug}_number_{$year}";
+	
+		return apply_filters( "wpo_wcpdf_{$this->slug}_store_name", $store_name, $this );
```

Comment:
just `_store_name` may be confusing, I would explicitly call it `_number_store_name` in the filter